src/Config/Parser/ConfigParserInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+/**
+ * Slim Framework (https://slimframework.com)
+ *
+ * @license https://github.com/slimphp/Slim-Console/blob/0.x/LICENSE.md (MIT License)
+ */
+
+declare(strict_types=1);
+
+namespace Slim\Console\Config\Parser;
+
+use InvalidArgumentException;
+use Slim\Console\Config\Config;
+
+interface ConfigParserInterface
+{
+    /**
+     * @return Config
+     *
+     * @throws InvalidArgumentException
+     */
+    public function parse(): Config;
```

Comment:
Is there a reason why you decided to pass the path to the constructor? We could avoid the whole abstraction by simply defining the interface to have the method 
```php
public function parse(string $path): Config;
```
and then use it like
```php
return (new PHPConfigParser())->parse($path);
```

Reply:
We could also avoid object instantiation and make that method static. 

I have one problem though, we need to inject `rootDir` through all of this so it can be passed into `Config::fromArray()` since it isn’t user defined but established through local detection. 

Still wondering what best approach would be

Reply:
> We could also avoid object instantiation and make that method static.

Yes. And if a user wants to add more formats, the `ConfigResolver` could be extended, right? Or do you think we should add some sort of config format repository (an array containing instances of config parsers which can be modified using some sort of `addConfigFormat` method)?

> I have one problem though, we need to inject rootDir through all of this so it can be passed into Config::fromArray() since it isn’t user defined but established through local detection.

If this is not user defined, then does `rootDir` even belong to `Config`? Or should that be part of the `App`?

But if it does, we could for example change `\Slim\Console\Config\ConfigResolver::resolve()` such that
```php
    return Config::fromEnvironment($this->rootDir);
```
and then `\Slim\Console\Config\Config::fromEnvironment()`
```php
    public static function fromEnvironment(string $rootDir): Config
    {
        return self::fromArray([
            'bootstrapDir' => getenv(self::SLIM_CONSOLE_BOOTSTRAP_DIR),
            'indexDir' => getenv(self::SLIM_CONSOLE_INDEX_DIR),
            'indexFile' => getenv(self::SLIM_CONSOLE_INDEX_FILE),
            'rootDir' => $rootDir,
            'sourceDir' => getenv(self::SLIM_CONSOLE_SOURCE_DIR),
            'commandsDir' => getenv(self::SLIM_CONSOLE_COMMANDS_DIR),
        ]);
    }
```
And in `\Slim\Console\Config\ConfigResolver::attemptParsingConfigFromFile()` we could write
```php
    PHPConfigParser::parse($path, $this->rootDir);
```
and then
```php
    public function parse(string $path, string $rootDir): Config
    {
        $parsed = require $path;

        if (!is_array($parsed)) {
            throw new InvalidArgumentException('Slim Console configuration should be an array.');
        }

        $parsed['rootDir'] = $rootDir;
        return Config::fromArray($parsed);
    }
```

Reply:
So one thing we could do is compute the paths outside of the `Config` object so we don’t have to inject it. I also feel we should rename `rootDir` to `basePath` and move that inside of the `App` component. It just makes things cleaner in my opinion. 

Then when we instantiate the `App` we can just inject the `basePath` in there. If commands need to compute paths from the config we could add a helper to merge the base path with any other specified path. 

What do you think @adriansuter?

Reply:
Great idea.

Do you think these helper methods should be part of the `App`? I mean
```php
class App extends SymfonyApplication
{
    /* ... */
    public function getSourceDir(): string
    {
        return $this->basePath . DIRECTORY_SEPARATOR . $this->config->getSourceDir();
    }
    /* ... */
}
```
and then
```php
abstract class AbstractCommand extends Command
{
    public function getApp(): App
    {
        $app = $this->getApplication();

        if ($app instanceof App === false) {
            throw new RuntimeException(get_class($app) . ' is not an instance of ' . App::class);
        }

        return $app;
    }
}
```
So a command can simply call
```php
    $this->getApp()->getSourceDir();
```

Reply:
Yeah I think this makes sense @adriansuter. I'm going to leave the implementation of those methods on the `App` for a different PR though. For now I'll write the tests for the `Config` and `ConfigResolver` objects.